src/calculateConfig.ts

```diff
@@ -0,0 +1,116 @@
+import snakeCase from 'lodash/snakeCase';
+import {
+  IntegrationExecutionContext,
+  IntegrationValidationError,
+} from '@jupiterone/integration-sdk-core';
+
+import {
+  DEFAULT_FINDINGS_SINCE_DAYS,
+  DEFAULT_SCANNED_SINCE_DAYS,
+} from './constants';
+
+import { CalculatedIntegrationConfig, IntegrationConfig } from './types';
+
+export function calculateConfig({
+  instance,
+  executionHistory,
+}: IntegrationExecutionContext<
+  IntegrationConfig
+>): CalculatedIntegrationConfig {
+  const config = instance.config;
+
+  const now = Date.now();
+  const lastSuccessfulExecutionTime =
+    executionHistory.lastSuccessful?.startedOn || 0;
+
+  const minScannedSinceDays = readPropertyAsNumberFromEnvOrConfig(
+    config,
+    'minScannedSinceDays',
+    DEFAULT_SCANNED_SINCE_DAYS,
+  );
+
+  const minScannedSinceTime = sinceDaysTime({
+    now,
+    sinceDays: minScannedSinceDays,
+  });
+
+  const minScannedSinceISODate =
+    lastSuccessfulExecutionTime > minScannedSinceTime
+      ? isoDate(lastSuccessfulExecutionTime)
+      : isoDate(minScannedSinceTime);
+
+  const maxScannedSinceISODate = isoDate(executionHistory.current.startedOn);
+
+  const minFindingsSinceDays = readPropertyAsNumberFromEnvOrConfig(
+    config,
+    'minFindingsSinceDays',
+    DEFAULT_FINDINGS_SINCE_DAYS,
+  );
+
+  const minFindingsSinceTime = sinceDaysTime({
+    now,
+    sinceDays: minFindingsSinceDays,
+  });
+
+  const minFindingsSinceISODate =
+    lastSuccessfulExecutionTime > minFindingsSinceTime
+      ? isoDate(lastSuccessfulExecutionTime)
+      : isoDate(minFindingsSinceTime);
+
+  const maxFindingsSinceISODate = isoDate(executionHistory.current.startedOn);
+
+  return {
+    ...config,
+
+    minScannedSinceDays,
+    minScannedSinceISODate,
+    maxScannedSinceISODate,
+
+    minFindingsSinceDays,
+    minFindingsSinceISODate,
+    maxFindingsSinceISODate,
+  };
+}
+
+function readPropertyFromEnv(propertyName: string): string | undefined {
+  const envName = snakeCase(propertyName).toUpperCase();
+  return process.env[envName];
+}
+
+function parsePropertyAsNumber(
+  propertyName: string,
+  value: string | undefined,
+): number | undefined {
+  if (!value) return undefined;
+
+  const numericValue = Number(value) || undefined;
+  if (!numericValue && !!value.trim()) {
+    throw new IntegrationValidationError(`Invalid ${propertyName}: ${value}`);
+  }
+
+  return numericValue;
+}
+
+function readPropertyAsNumberFromEnvOrConfig(
+  config: IntegrationConfig,
+  propertyName: keyof IntegrationConfig,
+  defaultValue: number,
+): number {
+  const envValue = readPropertyFromEnv(propertyName as string);
+  const configValue = config[propertyName as string];
+  return (
+    parsePropertyAsNumber(propertyName as string, envValue) ||
+    parsePropertyAsNumber(propertyName as string, configValue) ||
+    defaultValue
+  );
+}
```

Comment:
I see that this convention is used in Qualys, reading this property from _either_ the env or config. However, I would prefer to _only_ read this from the config, and always use either the config's value or the DEFAULT_VALUE defined in code. This cleaner interface will be especially notable when this code moves into the client calls.

Reply:
Done, but do let me know in case I misunderstood you (based on the new code you'll see)